url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/http/pools/ClientPool.java

```diff
@@ -0,0 +1,113 @@
+package stv6.http.pools;
+
+import java.util.LinkedList;
+import java.util.concurrent.Semaphore;
+
+import stv6.http.Client;
+
+/**
+ * A client pool should allow for multiple, threaded
+ * 	clients, without the risk of over-running Java's
+ * 	thread limit. Because, really, we probably don't
+ * 	need a separate thread for each client
+ * 
+ * For now, we'll be lazy
+ * 
+ * @author dhleong
+ *
+ */
+public class ClientPool implements ClientList {
+    
+    /**
+     * If we don't get data within 15 seconds, timeout
+     */
+	public static final long TIMEOUT_DELAY = 15000;
+
+    private final class ClientThread implements Runnable {
+
+	    private final boolean mRunning = true;
+		public String name;
+
+        @Override
+        public void run() {
+            while (mRunning) {
+                try {
+                    mClientsWaiting.acquire();
+                } catch (final InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)